odoo/models.py

```diff
@@ -3011,21 +3011,25 @@ def qualify(field):
                 res = 'pg_size_pretty(length(%s)::bigint)' % res
             return '%s as "%s"' % (res, col)
 
-        # selected fields are: 'id' followed by fields_pre
-        qual_names = [qualify(name) for name in [self._fields['id']] + fields_pre]
+        if fields_pre:
+            # selected fields are: 'id' followed by fields_pre
```

Comment:
Only the definition of `fields_pre` is useful before this `if` statement.
You can move the definition of `query` and `qualify` inside the `if` statement.

Reply:
Ok (moving also env, cr, param_ids accordingly)